### ISSUE TITLE
Issues with linking to gsl cblas

### ISSUE THREAD

**Issue by quadrikel** [mattpitkin/lintegrate#27]
Hi @mattpitkin,
this library seems to be exactly what I was looking for for my current project! Unfortunately, I cannot get it to run properly. I keep getting a linking error on my Mac:
`File "/usr/local/lib/python3.9/site-packages/lintegrate/__init__.py", line 3, in <module>
    from .lintegrate import *
ImportError: dlopen(/usr/local/lib/python3.9/site-packages/lintegrate/lintegrate.cpython-39-darwin.so, 2): Symbol not found: _cblas_caxpy
  Referenced from: /usr/local/opt/gsl/lib/libgsl.27.dylib
  Expected in: flat namespace
 in /usr/local/opt/gsl/lib/libgsl.27.dylib`

I also tried to run it on Ubuntu on a different machine, however I also ended up with an equivalent error:
` File "/home/myName/.local/lib/python3.8/site-packages/lintegrate-0.1.14-py3.8-linux-x86_64.egg/lintegrate/__init__.py", line 3, in <module>
    from .lintegrate import *
ImportError: /usr/local/lib/libgsl.so.27: undefined symbol: cblas_ctrmv`

It seems to me that linking to cblas does not work and it is looking in the wrong place...This happens already when I try a minimal example where I just import lintegrate.
I tried out a couple of things to fix this but nothing seems to help. Maybe you have encountered this issue before and have an idea of what might be going wrong. Would be great to use your library!

**Comment by mattpitkin** [comment]
Hi @quadrikel, thanks for your message. Have you just installed the package from PyPI with `pip install lintegrate` or did you install from a clone of the repository?

Unfortunately, I've had various issues in the past with compilation due to Cython, NumPy or GSL/cblas incompatibilities. I had thought that had stabilised and the current version worked, but obviously not!

One thing that should work for the moment is to install the package in a conda environment. E.g., [install miniconda](https://docs.conda.io/projects/miniconda/en/latest/miniconda-install.html) and then do:

```bash
# create new conda environment
conda create -n myenv python=3.9

# activate the environment
conda activate myenv

# install lintegrate from the conda-forge channel
conda install -c conda-forge lintegrate
```

This currently provides a working version of lintegrate on both a Linux and Mac.

I'll try and look into what the problems are for the other install pathways.

**Comment by mattpitkin** [comment]
I've created a new version of lintegrate that fixes a particular Cython build issues. I don't think this is directly related to your problem, but it may be worth given the new version a try.

The CI with the repo has successfully built and tested lintegrate on Ubuntu 20.04 and 22.04, along with Windows and Mac (although I'm not sure if it's an M1 architecture build). So, I'm not sure what's going on in your case. In general, I'd recommend the conda build unless you need lintegrate to be a requirement of your own package. If it's the latter case, we can try and dig a bit deeper into the problem.

**Comment by mattpitkin** [comment]
@quadrikel Just as a check, can you run the commands:

```bash
gsl-config --libs --cflags
```

on you Mac and Ubuntu machines and show what you get?

**Comment by quadrikel** [comment]
Hi @mattpitkin, thanks for all your help! To answer you questions: I could not install the package from PyPI, so I cloned the repo and installed manually (using scons and the setup.py as you suggest in the readme). I haven't tried the solution with Conda, yet. However, I found an easy fix that works on both my Mac and Ubuntu machine. I have added gslblas to the setup.py:

`libraries=[
                "gsl",
                "gslcblas",
            ],`

This seems to fix my issue.
